src/plugins/bfetch/public/batching/create_streaming_batched_function.ts

```diff
@@ -83,31 +78,56 @@ export const createStreamingBatchedFunction = <Payload, Result extends object>(
     maxItemAge = 10,
   } = params;
   const [fn] = createBatchedFunction<BatchedFunc<Payload, Result>, BatchItem<Payload, Result>>({
-    onCall: (payload: Payload) => {
+    onCall: (payload: Payload, signal?: AbortSignal) => {
       const future = defer<Result>();
       const entry: BatchItem<Payload, Result> = {
         payload,
         future,
+        signal,
+        done: false,
       };
       return [future.promise, entry];
     },
     onBatch: async (items) => {
       try {
-        let responsesReceived = 0;
-        const batch = items.map(({ payload }) => payload);
+        const promises: Array<Promise<void>> = [];
+        const abortController = new AbortController();
+
+        // Filter out and reject any items who's signal is already aborted
+        items = items.filter((item) => {
+          if (item.signal?.aborted) {
+            item.future.reject(new AbortError());
+          }
+          return !item.signal?.aborted;
+        });
+
+        // Prepare batch
+        const batch = items.map((item) => {
+          // Subscribe to reject promise on abort
+          rejectOnAbort(item);
+
+          // Track batch progress
+          promises.push(getDonePromise(item));
+
+          // Return payload to be sent
+          return item.payload;
+        });
+
+        // abort when all items were either resolved, rejected or aborted
+        Promise.all(promises).then(() => abortController.abort());
```

Comment:
Do I understand it correctly that:

1. Each item in a batch will come with optional `signal`, which can abort that specific item?
2. If all items are aborted than the whole batch should be aborted, too?

If that is the case, I think it is better to implement it as it was before with the counters. Putting back the `responsesReceived` counter and possibly introducing a counter for aborted items, like `numberOfAbortedItems`. And then use those counts to abort the whole batch when needed.

Reply:
You understood correctly and I thought so too at first, but the counters are problematic, because the same request can be received and then aborted, you'd need to be very careful not to count the same request twice towards both counters.

I think it's simpler and more straight forward with the implementation I did here.
Do you see any problems with my implementation, other than stlistically?

Reply:
I don't think it matters, as `fromStreamingXhr` makes sure not to apply abort to an already complete xhr request.
We could track each resolution separately, but I don't think it's worth it.